source/facter/3.0/release_notes.md

```diff
@@ -5,6 +5,103 @@ title: "Facter 3.0: Release Notes"
 
 This page documents the history of the Facter 3.0 series. (Previous version: [Facter 2.4 release notes](../2.4/release_notes.html).)
 
+Facter 3.0.2
+-----
+
+Released July 22, 2015.
+
+Shipped in puppet-agent version: 1.2.2.
+
+Facter 3.0.2 fixes several regressions in Facter 3.0.0 and implements new features to support legacy workflows.
+
+For JIRA issues related to Facter 3.0.2, see:
+- [Fixes for facter 3.0.2](https://tickets.puppetlabs.com/issues/?filter=15119)
+- [Introduced in facter 3.0.2](https://tickets.puppetlabs.com/issues/?filter=15118)
+
+### Feature: `--show-legacy` Flag Outputs Unstructured Facts
+
+Facter 3 outputs new, structured versions of many facts, such as `os`, and hides the equivalent unstructured legacy facts. In Facter 3.0.2, you can force Facter's output to include legacy facts by running it with the `--show-legacy` flag, providing a way to maintain legacy workflows while you convert them to the preferred `puppet facts`.
+
+- [FACT-1075](https://tickets.puppetlabs.com/browse/FACT-1075)
+
+### FIX: Improved Memory Usage on Launch
+
+Facter 3.0.0 consumes more memory than is necessary when starting `puppetserver`. Facter 3.0.2 significantly improves memory usage on launch, making `puppetserver` easier to start on low-spec systems.
+
+- [FACT-1083](https://tickets.puppetlabs.com/browse/FACT-1083)
+
+### FIX: Improved Parsing of Kernel-derived Major and Minor Operating System Version Facts
+
+In Facter 3.0.2, the `os` fact reports major and minor kernel release versions on operating systems that do not formally report OS release versions, such as OS X and Arch Linux.
+
+- [FACT-1056: Operatingsystem major release not resolved in OSX](https://tickets.puppetlabs.com/browse/FACT-1056)
+- [FACT-1073: Operatingsystem release facts not properly resolved in Arch Linux](https://tickets.puppetlabs.com/browse/FACT-1073)
+
+### REGRESSION FIX: Corrected Operating System Fact Reporting on Oracle Linux
+
+Facter 3.0.0 incorrectly reports Oracle Linux's `operatingsystem` and `osfamily` facts as `RedHat`. Facter 3.0.2 correctly reports `OracleLinux`.
+
+- [FACT-1134](https://tickets.puppetlabs.com/browse/FACT-1134)
+
+### REGRESSION FIX: Restored `xendomains` Fact
+
+Facter 3.0.2 reports the `xendomains` fact, which was unintentionally omitted from Facter 3.0.0.
+
+- [FACT-867](https://tickets.puppetlabs.com/browse/FACT-867)
+
+### REGRESSION FIX: Restored Hostname to the `fqdn` Fact
+
+When reporting the `fqdn` fact, Facter 3.0.0 doesn't report the hostname. Facter 3.0.2 restores the hostname to the fact.
```

Comment:
IIUC it was the other way around. It was *only* reporting the hostname, not including the domainname as well.

Also, it was only on some systems, not all. (Worth pointing that out so people don't look at release notes and wonder "funny, seemed to be working for me".)